### PR TITLE
Move brand config normalization from YAML to Ruby

### PR DESCRIPTION
**Pull request by delano**
## Summary

Move `BRAND_*` environment-variable handling out of the YAML/ERB layer and into Ruby (`Onetime::Config#normalize_brand`, run in `after_load`). This fixes YAML-significant characters — most visibly the leading `#` in hex color codes — being mangled during parsing, and hardens the YAML layer so operator-supplied brand values can't abort config load at boot.

The config **shape is unchanged**: same keys, same `BRAND_*` env vars — only the parsing mechanism moves to Ruby, so no Zod-schema or `DEPRECATIONS` changes are required. (The broader brand config-layer consolidation — collapsing `SITE_NAME`/`LOGO_URL`/`header.branding` into the `brand:` block — is tracked separately in #3606 and is **not** part of this PR.)

### Changes

1. **lib/onetime/config.rb** — added `normalize_brand()`, called during `after_load()`:
   - Maps each brand key to its `BRAND_*` env var and re-reads it in Ruby
   - Trims whitespace, converts blanks to `nil`
   - Coerces `button_text_light` to a real boolean (only an explicit `false`, whitespace-tolerant, disables it)
   - An env-set field always wins; a value set directly in YAML is preserved when the env var is unset

2. **etc/defaults/config.defaults.yaml** — interpolate each `BRAND_*` var as a JSON-quoted scalar (`&.to_json`) so YAML-significant values (hex `#`, a leading `&`/`*`/`!`, an embedded `: `, newlines) parse cleanly instead of dropping to `nil` or aborting the document at boot. `normalize_brand` remains the authority.

3. **src/apps/secret/conceal/HomepageContent.vue** — pass `primaryColor` from the identity store to `SecretForm` so the homepage CTA matches the branded logo color.

4. **src/shared/stores/bootstrapStore.ts** — use `structuredClone()` instead of a shallow spread in `state()` to stop nested defaults (e.g. `ui.header`) bleeding across Pinia instances in tests.

5. **Tests & docs** — `spec/unit/onetime/config/normalize_brand_spec.rb` (hex recovery, quote preservation, whitespace/blank→nil, YAML fallback, `button_text_light` coercion incl. whitespace + blank), a `load_spec.rb` regression proving a YAML-hostile brand value parses instead of aborting, and `docs/architecture/branding.md`.

### Review round (Copilot / Greptile)

- **Fixed** — `button_text_light` now strips before the `!= 'false'` comparison; a `' false '` value was silently enabling light text.
- **Fixed** — the JSON-quoted interpolation above closes a boot-crash vector where an operator brand value with YAML-significant characters raised `Psych::SyntaxError`/`AnchorNotDefined` before `normalize_brand` could run.
- **Declined (with rationale)** — `structuredClone` fallback: target is ES2022 (no legacy plugin), and a JSON deep-clone would drop the explicit `undefined` keys `DEFAULTS` needs for Pinia reactivity, whereas `structuredClone` preserves them.

## Related Issues

- Follow-up: #3606 (brand config-layer consolidation, deferred).
- Follow-up: #3610 (systemic ENV→YAML interpolation safety — the general class this PR's brand fix is one instance of).

## Test Plan

Unit specs cover all normalization scenarios plus the whitespace/blank `button_text_light` edge cases; a load spec guards YAML-parse safety against hostile brand values. Existing Pinia tests pass with the `structuredClone` isolation fix.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01MvR9x1zU2TBnmgQV43bD2p

### CODE REVIEW

File: lib/onetime/config.rb [hunk 1]
```diff
@@ -467,6 +472,62 @@ def after_load(loaded_config)
...
+        case brand['button_text_light']
+        when nil then nil
+        when true, false then brand['button_text_light']
+        else brand['button_text_light'].to_s != 'false'
+        end
+      elsif raw.strip.empty?
+        nil
+      else
+        raw != 'false'
+      end
```
[the remark this answers is not in the document]
Comment (delano) [reply lib/onetime/config.rb:545]: Fixed in 723670a. The `else` branch now compares `raw.strip != 'false'`, so `' false '` (or any surrounding whitespace) disables light text consistently with every other stripped field, instead of silently returning `true`. Added a regression spec for `'  false  '` → `false` and one for a whitespace-only value → `nil`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: etc/defaults/config.defaults.yaml [hunk 2]
```diff
@@ -414,23 +414,31 @@ site:
...
-  support_email: <%= ENV['BRAND_SUPPORT_EMAIL'].to_s.empty? ? 'null' : %Q{'#{ENV['BRAND_SUPPORT_EMAIL'].gsub("'", "''")}'} %>
-  corner_style: <%= ENV['BRAND_CORNER_STYLE'].to_s.empty? ? 'null' : %Q{'#{ENV['BRAND_CORNER_STYLE'].gsub("'", "''")}'} %>
-  font_family: <%= ENV['BRAND_FONT_FAMILY'].to_s.empty? ? 'null' : %Q{'#{ENV['BRAND_FONT_FAMILY'].gsub("'", "''")}'} %>
-  button_text_light: <%= ENV['BRAND_BUTTON_TEXT_LIGHT'].to_s.empty? ? 'null' : ENV['BRAND_BUTTON_TEXT_LIGHT'] != 'false' %>
-  logo_url: <%= ENV['BRAND_LOGO_URL'].to_s.empty? ? 'null' : %Q{'#{ENV['BRAND_LOGO_URL'].gsub("'", "''")}'} %>
-  favicon_url: <%= ENV['BRAND_FAVICON_URL'].to_s.empty? ? 'null' : %Q{'#{ENV['BRAND_FAVICON_URL'].gsub("'", "''")}'} %>
+  primary_color: <%= ENV['BRAND_PRIMARY_COLOR'] %>
+  product_name: <%= ENV['BRAND_PRODUCT_NAME'] %>
+  product_domain: <%= ENV['BRAND_PRODUCT_DOMAIN'] %>
+  support_email: <%= ENV['BRAND_SUPPORT_EMAIL'] %>
```
[the remark this answers is not in the document]
Comment (delano) [reply etc/defaults/config.defaults.yaml]: Good catch — and it was worse than mangling: it's a boot-crash vector. Fixed in 723670a. Each `BRAND_*` var is now interpolated as a JSON-quoted scalar (`ENV['BRAND_*']&.to_json`), so a realistic operator value — `& Co`, a leading `*`, an embedded `: `, or a newline — parses cleanly instead of raising `Psych::SyntaxError`/`AnchorNotDefined` before `normalize_brand` ever runs (the hex `#` survives too). An unset var still interpolates to nothing → YAML `nil`. `normalize_brand` remains the authority for trimming/blank→nil/boolean coercion. Added a `load_spec` regression that loads `'& Co: "special" *value'` and asserts the document parses instead of aborting.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/shared/stores/bootstrapStore.ts [hunk 3]
```diff
@@ -138,8 +138,12 @@ export const useBootstrapStore = defineStore('bootstrap', {
...
 
+  // structuredClone, not a shallow spread: DEFAULTS contains nested objects
+  // (e.g. ui.header) and a shallow copy would share those references across
+  // every store instance, letting one instance's $patch bleed into the next
+  // (notably across createTestingPinia instances in tests).
   state: (): BootstrapState => ({
-    ...DEFAULTS,
+    ...structuredClone(DEFAULTS),
     _initialized: false,
   }),
```
[the remark this answers is not in the document]
Comment (delano) [reply src/shared/stores/bootstrapStore.ts:148]: Leaving `structuredClone` as-is here, deliberately. The build targets ES2022 (`tsconfig` `target: es2022`, single non-legacy bundle, no `@vitejs/plugin-legacy` or browserslist floor), and every ES2022-era runtime ships `structuredClone` (Chrome 98+/FF 94+/Safari 15.4+/Node 17+). The `vite/modulepreload-polyfill` in `main.ts` is a build-mechanism shim for `<link rel=modulepreload>`, not a legacy-browser floor.

A JSON-based deep-clone fallback would also be *incorrect* here: `DEFAULTS` intentionally holds explicit `undefined` values so Pinia tracks those keys reactively (see the comment on the `DEFAULTS` block), and `JSON.parse(JSON.stringify(...))` drops `undefined` keys — reintroducing the exact reactivity gap that object exists to avoid, whereas `structuredClone` preserves them. If a legacy target is added later, the fallback would need a manual recursive clone that preserves `undefined`, not a JSON round-trip.

---
_Generated by [Claude Code](https://claude.ai/code)_
